xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/config/BankCodeFilter.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018-2018 adorsys GmbH & Co KG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.adorsys.xs2a.gateway.config;
+
+import de.adorsys.xs2a.gateway.service.Headers;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import java.io.IOException;
+import java.util.*;
+
+@Profile("oneBankMode")
+@Component
+public class BankCodeFilter implements Filter {
+
+    @Value("${one-bank-mode.code}")
+    private String bankCode;
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        if (bankCode != null && !bankCode.isEmpty()) {
+            MutableHttpServletRequest wrapper = new MutableHttpServletRequest((HttpServletRequest) request);
+            wrapper.addHeader(Headers.X_GTW_BANK_CODE, bankCode);
+            chain.doFilter(wrapper, response);
+        } else {
+            chain.doFilter(request, response);
+        }
+    }
+
+
+    public final class MutableHttpServletRequest extends HttpServletRequestWrapper {
+        // holds custom header and value mapping
+        private final Map<String, String> customHeaders;
+
+        MutableHttpServletRequest(HttpServletRequest request) {
+            super(request);
+            this.customHeaders = new HashMap<>();
+        }
+
+        void addHeader(String name, String value) {
+            this.customHeaders.put(name, value);
+        }
+
+
+        @Override
```

Comment:
Do we need these 3 overridden methods if we don't use them? Or it was made for future usages?

Reply:
It's not for us, it 's for everyone who used request object( filters, spring controllers, etc). I've spent two hours in deep spring debugging to realize that I should override one more method.
So the code is actual